compiler/src/iree/compiler/Codegen/Common/GPU/GPULowerToGlobalLoads.cpp

```diff
@@ -0,0 +1,265 @@
+        context, static_cast<gpu::MappingId>(mappingId)));
+  }
+  std::reverse(mapping.begin(), mapping.end());
+
+  Value subgroupId = rewriter.create<gpu::SubgroupIdOp>(loc, nullptr);
+  Value laneId = rewriter.create<gpu::LaneIdOp>(loc, nullptr);
+
+  // compute number of loads per thread
+  auto sourceType = cast<MemRefType>(copy.getOperand(0).getType());
+  MemRefType localType = cast<MemRefType>(copy.getOutputs().front().getType());
+
+  auto getGlobalGatherIndex = [&](Value sgIdVal, Value lIdVal,
+                                  Value indVar) -> Value {
+    auto zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    return rewriter.create<affine::AffineLinearizeIndexOp>(
+        loc, ValueRange{sgIdVal, indVar, lIdVal, zero},
+        ArrayRef<int64_t>{numSubgroups, numCopiesPerThread, subgroupSize[0],
+                          elementsPerCopy},
+        /*disjoint=*/true);
+  };
+
+  auto getSubgroupStoreBaseIndex = [&](Value sgIdVal, Value indVar) -> Value {
+    auto zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    return getGlobalGatherIndex(sgIdVal, zero, indVar);
+  };
+
+  // Build for loop skeleton:
+  scf::ForOp forOp = rewriter.create<scf::ForOp>(
+      loc, /*lb=*/rewriter.create<arith::ConstantIndexOp>(loc, 0),
+      /*ub=*/rewriter.create<arith::ConstantIndexOp>(loc, numCopiesPerThread),
+      /*steps=*/rewriter.create<arith::ConstantIndexOp>(loc, 1));
+
+  auto delinearizeIndex = [&](Value index, ArrayRef<int64_t> shape) {
+    return rewriter.create<affine::AffineDelinearizeIndexOp>(loc, index, shape)
+        .getMultiIndex();
+  };
+
+  // For loop body:
+  {
+    OpBuilder::InsertionGuard guard(rewriter);
+    rewriter.setInsertionPointToStart(forOp.getBody());
+    auto inductionVar = forOp.getInductionVar();
+    auto linearizedGatherIndices =
+        getGlobalGatherIndex(subgroupId, laneId, inductionVar);
+    auto delinearizedGlobalIndices =
+        delinearizeIndex(linearizedGatherIndices, sourceType.getShape());
+    auto linearizedBaseIndices =
+        getSubgroupStoreBaseIndex(subgroupId, inductionVar);
+    auto delinearizedLocalIndices =
+        delinearizeIndex(linearizedBaseIndices, localType.getShape());
+    rewriter.create<IREE::GPU::GlobalLoadDMAOp>(
+        loc, copy.getOperand(0), delinearizedGlobalIndices,
+        copy.getOutputs()[0], delinearizedLocalIndices);
+  }
+
+  // sync at the end of the loop across threads
+  rewriter.replaceOpWithNewOp<gpu::BarrierOp>(copy);
+  return success();
+}
+
+} // namespace
+
+static LogicalResult isEligibleForGlobalDMA(linalg::CopyOp copy) {
+  // source must be global address and target must be workgroup address.
+  auto sourceType = cast<MemRefType>(copy.getOperand(0).getType());
+  auto targetType = cast<MemRefType>(copy.getOutputs().front().getType());
+  if (!hasGlobalMemoryAddressSpace(sourceType)) {
+    LDBG("-- Op: "
+         << *copy
+         << "\n-- has source memory address space other than global.\n");
```

Comment:
I think we can break it into two `LDBG`, and it probably saves one line. (Same comment for below logs)

```cpp
LDBG("-- Op: " << *copy);
LDBG("-- has source memory address space other than global.");
```